src/runtime/objmodel.cpp

```diff
@@ -1048,6 +1072,31 @@ template <Rewritable rewritable> Box* typeLookup(BoxedClass* cls, BoxedString* a
         assert(attr->interned_state != SSTATE_NOT_INTERNED);
         assert(cls->tp_mro);
         assert(cls->tp_mro->cls == tuple_cls);
+
+
+
+        if (MCACHE_CACHEABLE_NAME(attr) && PyType_HasFeature(cls, Py_TPFLAGS_VALID_VERSION_TAG)) {
+            if (attr->hash == -1)
+                strHashUnboxed(attr);
+
+            /* fast path */
+            unsigned int h = MCACHE_HASH_METHOD(cls, attr);
+            if (method_cache[h].version == cls->tp_version_tag && method_cache[h].name == attr) {
+                if (rewrite_args) {
+                    RewriterVar* obj_saved = rewrite_args->obj;
+                    obj_saved->addAttrGuard(offsetof(BoxedClass, tp_flags), (intptr_t)cls->tp_flags);
+                    obj_saved->addAttrGuard(offsetof(BoxedClass, tp_version_tag), (intptr_t)cls->tp_version_tag);
```

Comment:
I think this does a 64-bit guard?  Maybe we should just make tp_version_tag 64 bits, which would let us avoid the wrap-around issue too :P